treeherder/push_health/push_health.py

```diff
@@ -69,19 +71,21 @@ def get_push_failures(push, option_map):
         test_key = '{}{}'.format(test_name, failure_line.job_guid)
         if test_name and test_key not in tests:
             job = failure_line.job_log.job
-            config = option_map[job.option_collection_hash]
+            config = clean_config(option_map[job.option_collection_hash])
+            platform = clean_platform(job.machine_platform.platform)
+            test_file = test_name.rsplit('/', 1)[-1]
             errors = []
             for step in failure_line.job_log.job.text_log_step.all():
                 for error in step.errors.all():
-                    if len(errors) < 5:
+                    if len(errors) < 5 and test_file in error.line:
```

Comment:
I think we need different logic here.

2 rules:
1) only look at the first 5
2) if line 1 has test_file, then ignore lines >1 that do not have test_file

counter = 0
found_test_file = false
for error in step.errors.all():
  if counter == 0 and test_file in error.line:
    found_test_file = true
  counter++
  if !found_test_file || counter >=5:
    break
  // here we keep only the top relevant lines
  errors.append(error.line)

Reply:
Interesting, ok.  I was concerned that we could have multiple tests failing for the same job, and so the first 5 log lines for the job **could** be consumed by the first test failure, and we'd miss the log lines for the second failure.  

Based the code you provided, am I right in thinking that subsequent test failures should not be considered independent wrt the log lines.  I know we touched on this in our chat, but I wasn't sure that was always going to be the case.

If I'm understanding that right, then, should I skip showing failed tests if it's not the first test that failed based on the log?

Reply:
in many cases failures are side effects of the first failure- when we currently sheriff we only consider the first failure to annotate.  while we will be overlooking many cases where real failures do happen, we have been doing that for a long time already.

Reply:
OK, great.  Thanks for clarifying.  :)  I'll make a note of this assumption in the code for "Future-Cam"  :)  Then I'll add a bit more logic here to suppress the second failed test to remove that noise from both the log lines and from the test failures in-general.

Thanks!!

Reply:
@jmaher : I updated the logic for this.  Would you take another look and see if it jibes with your thinking?  :)

Reply:
re-written as per our IRC conversation.